addons/l10n_jo_edi/models/account_move.py

```diff
@@ -0,0 +1,258 @@
+            'Content-Type': 'application/json',
+        }
+
+    def submit_to_jofotara(self):
+        self.ensure_one()
+        headers = self._build_jofotara_headers()
+        params = '{"invoice": "%s"}' % self.l10n_jo_edi_xml_attachment_id.datas.decode('ascii')
+
+        try:
+            response = requests.post(JOFOTARA_URL, data=str(params), headers=headers, timeout=50, verify=False)
+        except (requests.exceptions.Timeout, requests.exceptions.RequestException, requests.exceptions.HTTPError) as e:
+            raise UserError("Invalid request: %s" % e)
+
+        response_text = response.content.decode('utf-8')
+        if not response.ok:
+            raise UserError("Request failed: %s" % response_text)
+        dict_response = json.loads(response_text)
+        self.l10n_jo_edi_qr = str(dict_response.get('EINV_QR')) or ''
+
+    def _l10n_jo_edi_get_xml_attachment_name(self):
+        return f"{self.name.replace('/', '_')}_edi.xml"
+
+    def _l10n_jo_edi_send(self):
+        for invoice in self:
+            invoice.l10n_jo_edi_xml_attachment_id.res_field = False
+            invoice_xml = self.env["ir.attachment"].create(
+                {
+                    "res_model": "account.move",
+                    "res_id": invoice.id,
+                    "res_field": "l10n_jo_edi_xml_attachment_file",
+                    "name": invoice._l10n_jo_edi_get_xml_attachment_name(),
+                    "datas": base64.b64encode(invoice.env['account.edi.xml.ubl_21.jo']._export_invoice(invoice)),
+                }
+            )
+            invoice.invalidate_recordset(
+                fnames=[
+                    "l10n_jo_edi_xml_attachment_id",
+                    "l10n_jo_edi_xml_attachment_file",
+                ]
+            )
+
+            try:
+                invoice.submit_to_jofotara()
+                invoice.l10n_jo_edi_error = False
+                invoice.with_context(no_new_invoice=True).message_post(
+                    body=_("E-invoice (JoFotara) submitted successfully."),
+                    attachment_ids=invoice_xml.ids,
+                )
+            except (ValidationError, UserError) as error:
+                invoice.l10n_jo_edi_error = error
+                invoice.with_context(no_new_invoice=True).message_post(
+                    body=_("E-invoice (JoFotara) submission failed. %s", error),
+                    attachment_ids=invoice_xml.ids,
+                )
+                return error
+
+    @api.model
+    def create(self, vals):
+        records = super().create(vals)
+        for record in records:
+            if record.move_type in ['out_invoice', 'out_refund'] and record.company_id.country_id.code == 'JO':
+                record.l10n_jo_edi_uuid = uuid.uuid4()
+        return records
+
+
+class AccountMoveLine(models.Model):
+    _inherit = 'account.move.line'
+
+    def _get_conversion_rate(self):
+        return self.currency_rate or 1
```

Comment:
Use `currency_rate` then? Why creating a new method in a localization having such hightly generic name?